subsys/bluetooth/shell/bt.c

```diff
@@ -1074,6 +1093,25 @@ static int cmd_scan_filter_set_addr(const struct shell *sh, size_t argc,
 	return 0;
 }
 
+static int cmd_scan_filter_set_rssi(const struct shell *sh, size_t argc, char *argv[])
+{
+	int err = 0;
+	long rssi = shell_strtol(argv[1], 10, &err);
+
+	if (!err && (rssi < INT8_MAX) && (rssi > INT8_MIN)) {
```

Comment:
Shouldn't it be `<= MAX` and `>= MIN`, i.e. aren't the `MIN` and `MAX` values themselves valid? Btw, it's more conventional to put the exceptions into separate branches, rather than the normal code flow, so I'd reverse the condition and put the out of bounds handling into an if-branch.

Reply:
Agreed. Also consider using the `IN_RANGE` macro.

Reply:
I think it makes more sense to check `err` first (unless that's what you meant): because if err is not zero, then the `rssi` value might be bogus and it doesn't make sense to even check it.